js/game_state.js

```diff
@@ -308,4 +290,43 @@ var cGameState = function (level) {
 		if(finish_moves.length > 0)
 			return finish_moves[0];
 	}
-};
\ No newline at end of file
+};
+
+/**
+ * Copies all the cells from one board to another.
+ * @param {string[][]} from - Board to copy cells from.
+ * @param {string[][]} to  - Board to copy cells to.
+ */
+function cloneCells(from, to) {
+	for (var row = 0; row < 3; row++) {
+		for (var col = 0; col < 3; col++) {
+			cloneTo[row][col] = cloneFrom[row][col];
```

Comment:
The function parameters are not matching with the implementations.
`from `vs. `cloneFrom `and `to `vs `cloneTo`.  Can you please invoke them in your local and see for further issues?

Reply:
Sorry about this. I'm used to intelligent renaming which is doesn't really work with this. I'll fix it ASAP.